### PR TITLE
More .NET 6 changes

### PR DESCRIPTION
**Pull request by jbogard**

### CODE REVIEW

File: ContosoUniversity/Program.cs [hunk 1]
```diff
@@ -1,22 +1,70 @@
...
-}
+    else
+    {
+        app.UseDeveloperExceptionPage();
+    }
+
+    app.UseHttpsRedirection();
+    app.UseStaticFiles();
+
+    app.UseRouting();
```
Comment (adamjez) [inline ContosoUniversity/Program.cs:65]: According to the [docs](https://docs.microsoft.com/en-us/aspnet/core/migration/50-to-60](https://docs.microsoft.com/en-us/aspnet/core/migration/50-to-60#differences-between-the-aspnet-core-5-and-6-hosting-models)). It is no longer required:
> The endpoint routing middleware wraps the entire middleware pipeline, therefore there's no need to have explicit calls to UseRouting or UseEndpoints to register routes. UseRouting can still be used to specify where route matching happens, but UseRouting doesn't need to be explicitly called if routes should be matched at the beginning of the middleware pipeline.

File: ContosoUniversity/Program.cs [hunk 2]
```diff
@@ -1,22 +1,70 @@
...
-                });
+    if (!app.Environment.IsDevelopment())
+    {
+        app.UseExceptionHandler("/Error");
+        app.UseHsts();
     }
-}
+    else
+    {
+        app.UseDeveloperExceptionPage();
```
Comment (adamjez) [inline ContosoUniversity/Program.cs:59]: According to the [docs](https://docs.microsoft.com/en-us/aspnet/core/migration/50-to-60](https://docs.microsoft.com/en-us/aspnet/core/migration/50-to-60#differences-between-the-aspnet-core-5-and-6-hosting-models)). It is no longer required:
> In development mode, the developer exception page middleware is enabled by default.

**Review by adamjez — COMMENTED** [review]
Hey, I just came from your blog post. Some more clean up in middleware pipeline can be done.

File: ContosoUniversity/Program.cs (hunk 2 above)
Comment (jbogard) [reply ContosoUniversity/Program.cs:59]: Ah thanks! I created a new PR to incorporate this

**Pull request merged**